### ISSUE TITLE
slime-repl-set-package freezes clojure

### ISSUE THREAD

**Issue by edgargoncalves** [technomancy/swank-clojure#17]
I always thought namespaces should be changes automatically when switching buffers, or at least when compiling a file (with a `(ns ...)` form in it). But it doesn't. Also, using the `slime-repl-set-package` hangs my clojure session for good... I've asked (see http://stackoverflow.com/questions/3470999/how-do-i-connect-a-clojure-source-file-to-a-running-clojure-repl-on-emacs) if this was normal, and Isaac Hodes seems to share this behavior, so I've come up with this workaround http://gist.github.com/522706 . This is not optimal, by far, but it does help in the same way `slime-repl-set-package` was supposed to help :) 

Do you know what's happening, or do you have any thoughts on this? Thanks in advance!

**Comment by technomancy** [comment]
I can't reproduce it; I use C-c M-p every day with no trouble. Could you provide more details?

**Comment by edgargoncalves** [comment]
Hmm, sure - I explaind on the SO question, forgot to mention it here, sorry. I'm getting this hangs whenever I start the swank server via a java call, and use `M-x slime-connect`. It works when I use `lein swank`, though - what could I be doing wrong? Here's the embed fn I'm calling from java:

```
(ns offline.Embed
  (:require [swank.swank :as swank])
  (:gen-class
   :methods [#^{:static true} [startSwank [] void]]))

(defn -startSwank []
  (swank/start-repl))
```

Should I be doing something else?
(EDIT - I inadvertently clicked "comment and close" - how do i get this reopened? sorry...)

**Comment by edgargoncalves** [comment]
I'm glad to let you know that I may have had something messed up on my setup, but I finally have all things working. The hang up now only works on calls to `slime-sync-package-and-default-directory`, but `C-c M-p` is working fine. So the only positive thing I can give you now is perhaps an improvement suggestion - the code a provided on the gist I linked to detects the namespace from the buffer. It'd be a timesaver if `C-c M-p` suggested it automatically. I'm looking for the first occurrence of the `(ns ...)` form, but a file/directory logic can also be employed, I suppose - don't know which is faster/better.

Feel free to let this issue closed then (i guess my subconscience already knew this was solved :) ). Thanks!
